Docs/Verbali Esterni/VE_2024-12-19/content_01.tex

```diff
@@ -0,0 +1,64 @@
+Dalla risposta del proponente è emerso che le funzionalità di filtraggio sono in relazione di esclusività tra loro, quindi ne verrà selezionata solo una alla volta.\\ La visualizzazione del piano relativo al valore medio globale, invece, è indipendente dai filtri. Ci è stato suggerito di implementare una checkbox che permetta all'utente di decidere se visualizzare o meno il piano.
+\subsection{Test E2E}
+Pur trattandosi di un requisito opzionale ma gradito, e considerando che i membri del gruppo non ne avevano familiarità, abbiamo chiesto una spiegazione sui test E2E (End-to-End), in particolare su cosa siano e come possano essere implementati.\\
+Ci è stato spiegato che il testing E2E è fondamentale per garantire che l'applicazione funzioni correttamente dal punto di vista dell'utente finale. Ha spiegato che questo tipo di test permette di verificare come l'utente interagisce con l'applicazione e assicura che le funzionalità siano correttamente implementate. Inoltre, ha suggerito l'uso di strumenti come Selenium, Cypress e Playwright per automatizzare questi test, che registrano una serie di azioni dell'utente e le ripetono per verificare che l'impatto sia visibile in altre parti dell'applicazione. Un esempio pratico di test E2E potrebbe essere la verifica della lista di dataset, dove ci si aspetta che gli elementi siano visibili e che i dettagli vengano correttamente mostrati al clic.
+\subsection{Requisiti prestazionali e compatibilità con i browser}
+Abbiamo chiesto se fossero presenti anche dei requisiti prestazionali misurabili, come ad esempio il tempo di caricamento della pagina, e se dovessimo prendere in considerazione la retrocompatibilità con le versioni precedenti dei vari browser.\\
+Per quanto riguarda i requisiti prestazionali, ci è stato detto che non ci sono parametri particolari e definiti da rispettare. Tuttavia, è stato richiesto di immedesimarsi nell'utilizzatore dell'applicazione e porsi la domanda: \textit{"Quanto sarei disposto ad aspettare per il caricamento di questo dataset?"}.\\
+Proseguendo la discussione, si è giunti alla conclusione di basarsi sulla dimensione del dataset e ottenere un tempo di caricamento ragionevole, dell'ordine di grandezza dei secondi.\\
+A livello di compatibilità con i browser, ci è stato detto che è sufficiente garantire il funzionamento sull'ultima versione e assicurarsi che il prodotto sia utilizzabile nei browser più comuni.\\
+\subsection{Suddivisione del lavoro per PoC}
+Prima di terminare l'incontro, abbiamo richiesto al proponente qualche consiglio su come suddividere il lavoro di sviluppo del PoC.\\
+Ci è stato indicato che, a differenza dell'MVP, che richiede una divisione ben strutturata tra backend, frontend e interazione con il dataset, il PoC può essere sviluppato in parallelo, permettendo di testare singolarmente le tecnologie proposte per la realizzazione del progetto.\\
```

Comment:
Ci è stato indicato che, a differenza dell'MVP (il quale richiede una divisione ben strutturata tra backend, frontend e interazione con il dataset), il PoC può essere sviluppato in parallelo, permettendo di testare singolarmente le tecnologie proposte per la realizzazione del progetto.